### DISCUSSION TITLE
Ability to add class to {encode}

### DISCUSSION THREAD

**Discussion by mcfarlandonline** [ExpressionEngine/ExpressionEngine#3218] [Feature Requests]
I wonder how easy/difficult it would be to add a class parameter to the {encode} Global Variable? For example: 

> {encode="email@asitedomain.com" title="Email Me" class="nr-button nr-button-blue"}

This would reduce the need for a lot of extra CSS in our websites, and simplify training for our clients.

**Comment by intoeetive** [comment]
Checked and this seems to require rather big update in related functions. Listing as feature request however

**Reply by mcfarlandonline** [reply]
Thank you! And just to clarify, the hope for me would be that injected classes get applied to the anchor/<a> tag that is generated from this. Right now, I have to style these links like this: 

> <span class="nr-button nr-button-blue">{encode="[email@asitedomain.com](mailto:email@asitedomain.com)" title="Email Me"}</span>

Which requires duplicate CSS that adds up overtime. As a small example:  
```
.nr-button, .nr-button:visited, 
.nr-button a, .nr-button a:visited {
   color: white;
   background: black;
}
.nr-button-blue, .nr-button-blue:visited,
.nr-button-blue a, .nr-button-blue a:visited {
   background: blue;
} 
```

If you know of a better way of handling this, I'm all ears! 

Thanks.

**Comment by intoeetive** [comment]
Another request for this: https://github.com/ExpressionEngine/ExpressionEngine/issues/4469

**Comment by nep** [comment]
Would also love this.
